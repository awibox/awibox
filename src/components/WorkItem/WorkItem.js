import React from 'react';
import PropTypes from 'prop-types';
// Styles
import styles from './WorkItem.scss';

const WorkItem = (props) => {
  const { data } = props;
  return (
    <div className={styles.workItem}>
      <div className={styles.position}>{data.position}</div>
      <div className={styles.company}>{data.company}</div>
      <div className={styles.description}>{data.description}</div>
    </div>
  );
};

WorkItem.propTypes = {
  data: PropTypes.any,
};

export default WorkItem;
