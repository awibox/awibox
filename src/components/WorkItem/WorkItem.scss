@import '../../styles/variables';

.workItem {

}
.company {
  font-weight: bold;
  color: $orange;
}
.position {
  font-size: 18px;
  line-height: 18px;
  margin-bottom: 5px;
}
.location {
  margin-bottom: 5px;
  font-size: 13px;
  color: rgba(0,0,0,0.4);
}
.description {
  line-height: 19px;
  font-size: 13px;
  color: rgba(0,0,0,0.6);
  ul {
    margin-top: 5px;
    margin-bottom: 5px;
    margin-left: 15px;
  }
}
