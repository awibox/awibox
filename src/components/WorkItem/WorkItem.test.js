import React from 'react';
import { shallow } from 'enzyme';
import WorkItem from './WorkItem';

const mockData = {
  position: 'Engineer',
  company: 'Company',
  location: 'Moscow',
  description: 'Description',
};

describe('Timeline', () => {
  it('should render correctly', () => {
    const component = shallow(<WorkItem title='Test timeline' data={mockData} />);
    expect(component).toMatchSnapshot();
  });
});
