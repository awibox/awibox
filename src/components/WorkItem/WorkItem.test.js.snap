// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should render correctly 1`] = `
<div>
  <div>
    Engineer
  </div>
  <div>
    Company
  </div>
  <div
    className="description"
  >
    Description
  </div>
</div>
`;
